Extract setup methods from install_tasks

ondenman opened the pull request:
This PR is in preparation of https://github.com/everypolitician/scraper_test/pull/16

`install_tasks` has grown quite long. #16 will add further lines of code to the method but doing so will cause it to fail rubocop's line count check.

This PR extracts the `Results` class which takes a yaml test file and returns the actual and expected data.

tmtmtmtm reviewed (changes requested):
Factoring out dependencies into private sub-functions doesn't really seem like the best way to decompose this. 

As you're operating within `rake`, if you did want to split out 'require_dependencies', for example, that should, in turn, be _another_ rake task, that `install_tasks` depends on. 

Or, you could make it depend on `configure_vcr`, which in turns depends on `require_dependencies`. (Though if you're taking that approach, you should make sure to bundle *all* the VCR configuration into that — at the moment you've left the cleanup behind.) 

That approach gets a little complicated here, though, because we need to be careful with task naming. I haven't looked into whether there's a standard `rake` approach to that, though one route would probably be through use of a namespace.

However, I suspect a much better approach generally is to factor out the code that actually runs the test for a given YAML file, and leave the raw `task` function just for installing that ability into the caller.

That seems to be a more natural separation of concerns; will make it simpler to modify how the test-running code works (and in turn to decompose _that_); and make it much simpler to test the test-running in isolation from `rake`.

Commit:
Extract Results class

Commit:
Test ScraperTest::Results#expected

Commit:
Add minitest-around as development dependency

This is being added in preparation of using VCR in a test.

Commit:
Add scraped as development dependency

This is being added in preparation of testing Scraper_Test::Results. Scraper_Test::Results takes a yaml file which references a Scraped class to be tested. For testing purposes, a dummy Scraped class will be created in the test file.

Commit:
Set up VCR for tests

Commit:
Test SraperTest::Results#actual

ondenman commented:
I'm not sure if it's necessary but I've created a branch with a more suitable name and have opened a new PR. https://github.com/everypolitician/scraper_test/pull/18
